src/patches.js

```diff
@@ -6,50 +6,92 @@ export function generatePatches(state, basePath, patches, inversePatches) {
         : generateObjectPatches(state, basePath, patches, inversePatches)
 }
 
-function generateArrayPatches(state, basePath, patches, inversePatches) {
-    const {base, copy, assigned} = state
-    const minLength = Math.min(base.length, copy.length)
+function _generateArrayPatches(
+    base,
+    copy,
+    assigned,
+    basePath,
+    patches,
+    inversePatches
+) {
+    if (copy.length < base.length) {
+        return _generateArrayPatches(
+            copy,
+            base,
+            assigned,
+            basePath,
+            inversePatches,
+            patches
+        )
+    }
 
-    // Look for replaced indices.
-    for (let i = 0; i < minLength; i++) {
-        if (assigned[i] && base[i] !== copy[i]) {
-            const path = basePath.concat(i)
-            patches.push({op: "replace", path, value: copy[i]})
-            inversePatches.push({op: "replace", path, value: base[i]})
-        }
+    const delta = copy.length - base.length
+
+    let start = 0
```

Comment:
This needs a comment. What is this the `start` of?

Something like `// Find the first changed index.`